### ISSUE TITLE
lesson 2 decoder impl output to linear

### ISSUE THREAD

**Issue by iyersathya** [bentrevett/pytorch-seq2seq#19]
//feed, embedding+context and hidden state to RNN.       
output, hidden = self.rnn(emb_con, hidden)

### Shouldn't the line below be concat (embedding + output + context) instead of (embedding + hidden + context) ?  Calculated hidden vector by RNN at time step t  will be fed to next rnn at t+1 anyway.

**output = torch.cat((embedded.squeeze(0), _hidden.squeeze(0)_, context.squeeze(0)), 
                           dim = 1)**
        
prediction = self.out(output)

**Comment by bentrevett** [comment]
It can be either `output` or `hidden` here. Both of these tensors are exactly the same. You can test this with: `assert torch.all(torch.eq(output, hidden))`

**Comment by iyersathya** [comment]
@bentrevett  Thank you.
